akka-projection-core/src/main/scala/akka/projection/ProjectionSettings.scala

```diff
@@ -131,12 +135,29 @@ private[akka] class ProjectionSettingsImpl(
       randomFactor: Double = randomFactor,
       maxRestarts: Int = maxRestarts,
       saveOffsetAfterEnvelopes: Int = saveOffsetAfterEnvelopes,
-      saveOffsetAfterDuration: FiniteDuration = saveOffsetAfterDuration): ProjectionSettings =
+      saveOffsetAfterDuration: FiniteDuration = saveOffsetAfterDuration,
+      recoveryStrategy: HandlerRecoveryStrategy = recoveryStrategy): ProjectionSettings =
     new ProjectionSettingsImpl(
       minBackoff,
       maxBackoff,
       randomFactor,
       maxRestarts,
       saveOffsetAfterEnvelopes,
-      saveOffsetAfterDuration)
+      saveOffsetAfterDuration,
+      recoveryStrategy)
+}
+
+private object RecoveryStrategyConfig {
+  def fromConfig(config: Config): HandlerRecoveryStrategy = {
+    val strategy = config.getString("strategy")
+    val retries = config.getInt("retries")
+    val retryDelay = config.getDuration("retry-delay", MILLISECONDS).millis
+
+    strategy match {
+      case "fail"           => HandlerRecoveryStrategy.fail
+      case "skip"           => HandlerRecoveryStrategy.skip
+      case "retry-and-fail" => HandlerRecoveryStrategy.retryAndFail(retries, retryDelay)
+      case "retry-and-skip" => HandlerRecoveryStrategy.retryAndSkip(retries, retryDelay)
+    }
+  }
```

Comment:
I'm not convinced we should support configuring the default recovery strategy. That can always be `HandlerRecoveryStrategy.fail` and errors are handled by the full restart mechanism.

Reply:
@patriknw, do you mean not config at all and only change it programmatically? 

I personally favour programmatically, but why different approaches? If it's possible to change the backoff and the save offset frequency, why not the recovery?

Reply:
I’m thinking that we will have other more advanced recover strategies in the future that can’t be expressed in config.

Also recovery completely changes the behavior compared to frequency settings. Difficult to reason about the code if such changed behavior is hidden in config.

Reply:
> Difficult to reason about the code if such changed behavior is hidden in config.

It would be the user's config though, since our default would be `fail`.